### DISCUSSION TITLE
Confused by deeply nested vs immediately nested field arrays

### DISCUSSION THREAD

**Discussion by terebentina** [react-hook-form/react-hook-form#4042] [Q&A] [answered]
Hello,

I have a form where the shape of the object is like this: `outerArray[].someObj.nestedArray[].field` and I am running into some strange behaviour when removing items from the nested array.
To see the problem in action I created 2 code sandboxes:
- one with an object shape of `outerArray[].nestedArray[].field` which works fine (https://codesandbox.io/s/rhf-immediately-nested-arrays-wsru5?file=/src/App.js)
- another one with `outerArray[].someObj.nestedArray[].field` which doesn't work (https://codesandbox.io/s/rhf-deeply-nested-arrays-jq05n?file=/src/App.js)

The rest of the code is the same between the 2 CSBs.
To reproduce the problem click on one of the `remove nested item` buttons in the second example. There are even some errors in the console when you do this. Like I said, doing the same in the first example works perfectly with the same code.

A few notes which might be relevant:
- my form has to run with `shouldUnregister: false`
- out of curiosity I tried with `shouldUnregister: true` and this removes the item from the nested array but it still logs a warning in the console: `useFieldArray fieldValues contain the keyName "id" which is reserved for use by useFieldArray`. There's no id field in my data, it's added by `useFieldArray` but somehow it gets confused about this. Anyway, like I said, I have to run with `shouldUnregister: false` so this was just a test.
- the values of the fields are set with `defaultValues` in `useForm` but I also set the input in the field array to `defaultValue={field.value}` as specified in the docs.
- I use `FormProvider`.
- I checked the official "Array of Field Array" example from https://github.com/react-hook-form/react-hook-form/tree/master/examples but that's not using deeply nested field arrays as in my second example.

I am out of ideas for things to try atm so any help is appreciated.

**Comment by bluebill1049** [comment, marked as answer]
hey @terebentina, thanks for the detailed question and context around this. This is not going to work with v6 as the design was intended just made for an array of array fields, not chaining objects than with another field array. However, this is getting addressed in v7: https://codesandbox.io/s/rhf-deeply-nested-arrays-forked-4xf8e?file=/src/App.js:1518-1530 I would avoid the deep nested object of an array with v6.

**Comment by terebentina** [comment]
Oh, man! I am at the same time sad that it's not doable with v6 and excited about the imminent v7 release! It is imminent, isn't it? 😁 
Here's one crazy thing: if I `reset(getValues())` the form then...it works!
https://codesandbox.io/s/rhf-deeply-nested-arrays-with-reset-lbu3k?file=/src/Form.js:730-748

**Reply by bluebill1049** [reply]
good that you find a workaround, actually you can probably use `setValue('test', [])` as well.
